service-capabilities-update/src/main/java/org/oskari/capabilities/UpdateCapabilitiesJob.java

```diff
@@ -1,278 +1,187 @@
 
     private final OskariLayerService layerService;
     private final CapabilitiesCacheService capabilitiesCacheService;
-    private final int maxAge;
 
     public UpdateCapabilitiesJob() {
         this(new OskariLayerServiceIbatisImpl(),
-                new CapabilitiesCacheServiceMybatisImpl(),
-                PropertyUtil.getOptional(PROP_MAX_AGE, 0));
+                new CapabilitiesCacheServiceMybatisImpl());
     }
 
     public UpdateCapabilitiesJob(OskariLayerService layerService,
-            CapabilitiesCacheService capabilitiesService, int maxAge) {
+            CapabilitiesCacheService capabilitiesService) {
         this.layerService = layerService;
         this.capabilitiesCacheService = capabilitiesService;
-        this.maxAge = maxAge;
     }
 
     @Override
     public void execute(Map<String, Object> params) {
-        List<OskariLayer> layers = getValidLayers();
-        sortLayersByUrlTypeVersion(layers);
-        updateLayersGroupAtATime(layers);
-    }
-
-    protected List<OskariLayer> getValidLayers() {
-        // Filter out layers with invalid types
-        return layerService.findAll().stream()
+        LOG.info("Starting UpdateCapabilitiesJob");
+        layerService.findAllWithPositiveUpdateRateSec().stream()
                 .filter(layer -> canUpdate(layer.getType()))
-                .collect(Collectors.toCollection(ArrayList::new));
-    }
-
-    protected boolean canUpdate(String type) {
-        return OskariLayer.TYPE_WMTS.equals(type);
+                .filter(layer -> shouldUpdate(layer))
+                .collect(groupingBy(layer -> new UrlTypeVersion(layer)))
+                .forEach((utv, layers) -> updateCapabilities(utv, layers));
     }
 
-    protected void sortLayersByUrlTypeVersion(List<OskariLayer> layers) {
-        layers.sort(new Comparator<OskariLayer>() {
-            @Override
-            public int compare(OskariLayer o1, OskariLayer o2) {
-                int c = o1.getSimplifiedUrl(true).compareTo(o2.getSimplifiedUrl(true));
-                if (c != 0) {
-                    return c;
-                }
-                c = o1.getType().compareTo(o2.getType());
-                if (c != 0) {
-                    return c;
-                }
-                return o1.getVersion().compareTo(o2.getVersion());
-            }
-        });
+    protected static boolean canUpdate(String type) {
+        switch (type) {
+        case OskariLayer.TYPE_WMS:
+        case OskariLayer.TYPE_WMTS:
+            return true;
+        default:
+            return false;
+        }
     }
 
-    protected void updateLayersGroupAtATime(List<OskariLayer> layers) {
+    protected static boolean shouldUpdate(OskariLayer layer) {
+        Date lastUpdated = layer.getCapabilitiesLastUpdated();
+        if (lastUpdated == null) {
+            LOG.debug("Should update layer:", layer.getId(), "last updated unknown and update rate is positive");
```

Comment:
Update rate isn't checked here, but I guess if last update is null we should always update. Maybe change the message a bit?

Reply:
True, the message is a bit misleading, the check is done beforehand in layerService.findAllWithPositiveUpdateRateSec()